packages/webapp/src/containers/Finances/useTransactions.js

```diff
@@ -0,0 +1,215 @@
+  return expenses
+    .filter(
+      (expense) =>
+        (!dateFilter ||
+          (moment(expense.expense_date).isSameOrAfter(dateFilter.startDate, 'day') &&
+            moment(expense.expense_date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+        (!expenseTypeFilter || expenseTypeFilter?.includes(expense.expense_type_id)) &&
+        expense.value > 0,
+    )
+    .map((expense) => {
+      const expenseType = expenseTypes.find(
+        (expenseType) => expenseType.expense_type_id === expense.expense_type_id,
+      );
+      return {
+        icon: expenseType?.farm_id ? 'CUSTOM' : expenseType?.expense_translation_key,
+        date: expense.expense_date,
+        transactionType: transactionTypeEnum.expense,
+        typeLabel: expenseType?.farm_id
+          ? expenseType?.expense_name
+          : i18n.t(`expense:${expenseType?.expense_translation_key}`),
+        amount: -roundToTwoDecimal(expense.value),
+        note: expense.note,
+      };
+    });
+};
+
+const buildRevenueTransactions = ({
+  sales,
+  revenueTypes,
+  cropVarieties,
+  dateFilter,
+  revenueTypeFilter,
+}) => {
+  const filteredSales = sales.filter(
+    (sale) =>
+      (!dateFilter ||
+        (moment(sale.sale_date).isSameOrAfter(dateFilter.startDate, 'day') &&
+          moment(sale.sale_date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+      (!revenueTypeFilter || revenueTypeFilter?.includes(sale.revenue_type_id)),
+  );
+  const revenueItems = mapSalesToRevenueItems(filteredSales, revenueTypes, cropVarieties);
+
+  return revenueItems.map((item) => {
+    const revenueType = revenueTypes.find(
+      (revenueType) => revenueType.revenue_type_id == item.sale.revenue_type_id,
+    );
+    return {
+      icon: revenueType?.farm_id ? 'CUSTOM' : revenueType?.revenue_translation_key,
+      date: item.sale.sale_date,
+      transactionType: transactionTypeEnum.revenue,
+      typeLabel: revenueType?.farm_id
+        ? revenueType?.revenue_name
+        : i18n.t(`revenue:${revenueType?.revenue_translation_key}`),
+      amount: item.totalAmount,
+      note: item.sale.customer_name,
+      items: item.financeItemsProps,
+    };
+  });
+};
+
+export const buildTransactions = ({
+  sales,
+  tasks,
+  expenses,
+  expenseTypes,
+  revenueTypes,
+  taskTypes,
+  cropVarieties,
+  users,
+  dateFilter,
+  expenseTypeFilter,
+  revenueTypeFilter,
+}) => {
+  const transactions = [
+    ...buildLabourTransactionsFromTasks({ tasks, taskTypes, users, dateFilter, expenseTypeFilter }),
+    ...buildExpenseTransactions({ expenses, expenseTypes, dateFilter, expenseTypeFilter }),
+    ...buildRevenueTransactions({
+      sales,
+      revenueTypes,
+      cropVarieties,
+      dateFilter,
+      revenueTypeFilter,
+    }),
+  ];
+
+  const sortedTransactions = transactions.sort(getComparator('desc', 'date'));
+
+  return sortedTransactions;
+};
+
+const useTransactions = ({ dateFilter, expenseTypeFilter, revenueTypeFilter }) => {
```

Comment:
Sorry @antsgar, I have a request that I should have made earlier...
![Screenshot 2023-10-24 at 3 17 58 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/fe346254-b6d9-4a12-a978-55507cd300be)
I think we need `sale_id` and `farm_expense_id` to navigate to the detail page of each transaction. Would you be able to add them?